src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java

```diff
@@ -39,33 +42,81 @@ public NewHandler(IndentationCheck indentCheck,
                       DetailAST ast,
                       AbstractExpressionHandler parent) {
         super(indentCheck, "operator new", ast, parent);
+        mainAst = ast;
     }
 
     @Override
     public void checkIndentation() {
-        final DetailAST type = getMainAst().getFirstChild();
-        if (type != null) {
-            checkExpressionSubtree(type, getIndent(), false, false);
+        // if new is on the line start and it is not the part of assignment.
+        if (isOnStartOfLine(mainAst)
+                && !isNewKeywordFollowedByAssign()) {
+            final int columnNo = expandedTabsColumnNo(mainAst);
+            final IndentLevel level = getIndentImpl();
+
+            if (columnNo < level.getFirstIndentLevel()) {
+                logError(mainAst, "", columnNo, level);
+            }
+        }
+
+        final DetailAST firstChild = mainAst.getFirstChild();
+        if (firstChild != null) {
+            checkExpressionSubtree(firstChild, getIndent(), false, false);
         }
 
-        final DetailAST lparen = getMainAst().findFirstToken(TokenTypes.LPAREN);
+        final DetailAST lparen = mainAst.findFirstToken(TokenTypes.LPAREN);
         checkLeftParen(lparen);
     }
 
+    @Override
+    public IndentLevel getSuggestedChildIndent(AbstractExpressionHandler child) {
+        final int offset;
+        if (child.getMainAst().getType() == TokenTypes.OBJBLOCK) {
+            offset = getBasicOffset();
+        }
+        else {
+            offset = getLineWrappingIndent();
+        }
+        return new IndentLevel(getIndent(), offset);
+    }
+
     @Override
     protected IndentLevel getIndentImpl() {
-        final IndentLevel result;
+        IndentLevel result;
         // if our expression isn't first on the line, just use the start
         // of the line
-        if (getLineStart(getMainAst()) == getMainAst().getColumnNo()) {
+        if (getLineStart(mainAst) == mainAst.getColumnNo()) {
             result = super.getIndentImpl();
+
+            if (isNewKeywordFollowedByAssign()) {
+                result = new IndentLevel(result, getLineWrappingIndent());
+            }
         }
         else {
-            result = new IndentLevel(getLineStart(getMainAst()));
+            result = new IndentLevel(getLineStart(mainAst));
         }
+
         return result;
     }
 
+    /**
+     * Is NewKeyword is placed after an Assignment.
+     *
+     * @return is new keyWord is follwed by Assignment.
```

Comment:
`Is NewKeyword is placed after an Assignment.`
This is a bit confusing.  Maybe `Checks if the 'new' keyword is followed by an assignment`?

`@return is new keyWord is follwed by Assignment.` -> `@return true if new keyword is followed by assignment`

Reply:
Also, travis failure is from `follwed`.  As @romani has said, spelling check CI is the most difficult of all to pass :)

Reply:
>  spelling check CI is the most difficult of all to pass :)

Well said ;)

Reply:
Done

Reply:
Resolved.